packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+
+**⁉️ NOTE**: Publishing a new version shouldn't be done if the only change since it was las published is an `Upgrade` of dependency(ies).
+
+Let's imagine we have `packageB`, v1.5.0, that depends on `packageA`, v1.0.5. The following are some scenarios of what should be done:
+
+### No public changes
+
+* `packageA` adds support for something new → `New: Support for this new thing`.
+* `packageB` needs `packageA`'s new feature to fix something that was broken → `Fix: This nasty bug`.
+
+The release script will:
+* release `packageA` with a `minor` bump: version 1.1.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 1.1.0`. Because the commit history has `Fix` and `Upgrade`, `packageB` will be bumped to 1.5.1 (`Fix` == `Upgrade`).
+
+Let's use a more concrete example:
+
+`utils-create-server` switches their API to spawn a new process per server and makes all its API `async`. This is a breaking change so it should be noted as `Breaking: Make API async`. The release script will bump it to the next major version, so if previously it was `3.2.1`, it will be update to `4.0.0`. At the same time we do this breaking change we want to update the packages that depend on it:
+
+| Dependency |
+|------------|
+| @hint/connector-chrome |
+| @hint/connector-jsdom |
+| @hint/extension-browser |
+| @hint/utils-connector-tools |
+| @hint/utils-tests-helpers |
+
+The interaction in each package with `utils-create-server` needs to be updated, but nothing changes externally so the commit to make those changes should be `Chore: Update to new utils-create-server interface`. Let's pick `utils-tests-helpers` and assume the current version is `2.0.1` and its `package.json` could look like:
+
+```json
+{
+    ...
+    "dependencies": {
+        ...
+        "@hint/utils-create-server": "^3.2.1"
+        ...
+    },
+    "version": "2.0.1"
+}
+```
+
+After the release script it will look like:
+
+```json
+{
+    ...
+    "dependencies": {
+        ...
+        "@hint/utils-create-server": "^4.0.0"
+        ...
+    },
+    "version": "2.0.2"
+}
+```
+
+This is fine because of the way `semver` works. Version `2.0.1` and prior will download the latest `3.2.x` version of `@utils/create-test-server`. Version `2.0.2` and later will download `4.0.x`.
+
+Now that `@hint/utils-test-server` has been updated, what happens to all the other packages? The dependency will be updated but no version will be published because there will be only one `Upgrade` commit in the history. Those packages have a dependency such as:
+
+```json
+"@hint/utils-test-server": "^2.0.1"
+```
+
+When someone install the package from a clean cache version `2.0.2` will be downloaded among the latest `@hint/utils-create-server"` which is fully compatible with the current code.
```

Comment:
Also, probabaly good idea to add a few commas to make fragment the sentence a bit

When someone installs the package from a clean cache, version `2.0.2` will be downloaded among the latest `@hint/utils-create-server"`, which is fully compatible with the current code.